source/opt/optimizer.cpp

```diff
@@ -427,6 +427,15 @@ bool Optimizer::RegisterPassFromFlag(const std::string& flag) {
     }
   } else if (pass_name == "loop-peeling") {
     RegisterPass(CreateLoopPeelingPass());
+  } else if (pass_name == "loop-peeling-threshold") {
+    int factor = (pass_args.size() > 0) ? atoi(pass_args.c_str()) : 0;
+    if (factor != 0) {
```

Comment:
What happens if I pass -10?

Reply:
Good catch, though different bug.  This code is simply restoring code I had inadvertently deleted.  I'll fix on a separate PR.  I don't want to change existing behaviour in this PR.

I've created #1817 to track.

Reply:
On second thought,  the PR is fixing other bugs already.  Might as well Fix All The Bugs.